systemd/hyprland-session.service.in

```diff
@@ -1,16 +1,21 @@
 [Unit]
 Description=Hyprland - Tiling compositor with the looks
 Documentation=man:Hyprland(1)
-BindsTo=graphical-session.target
-Before=graphical-session.target
+
+BindsTo=hyprland-session.target
+Before=hyprland-session.target
+
 Wants=xdg-desktop-autostart.target
-Wants=graphical-session-pre.target
-After=graphical-session-pre.target
+
+Wants=hyprland-session-pre.target
+After=hyprland-session-pre.target
 
 [Service]
 Type=notify
-ExecStart=@PREFIX@/@BINDIR@/Hyprland
-ExecStop=@PREFIX@/@BINDIR@/hyprctl dispatch exit
+ExecStart=@BINDIR@/Hyprland
+ExecStop=@BINDIR@/hyprctl dispatch exit
```

Comment:
This breaks using the current flake:

```
...
[Service]
Type=notify
ExecStart=bin/Hyprland
ExecStop=bin/hyprctl dispatch exit
ExecStopPost=systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
OnFailure=systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
OnSuccess=systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
...
```

Reply:
We're scrapping the idea anyway.

Reply:
Hm, that's odd, bindir is set in the mesonFlags: `--bindir=/nix/store/81g3gcxh6ca0gk46r6p7zqna2fb3svd7-hyprland-0.44.0+date=2024-11-03_dirty/bin`

Reply:
Yes, after I built the package I got this

```ini
[Service]
Type=notify
ExecStart=/nix/store/hsx2dwp9agi54dvkkp7rhr9wgvar6d4h-hyprland-0.44.0+date=2024-11-03_403cd66/bin/Hyprland
ExecStop=/nix/store/hsx2dwp9agi54dvkkp7rhr9wgvar6d4h-hyprland-0.44.0+date=2024-11-03_403cd66/bin/hyprctl dispatch exit
ExecStopPost=systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
Restart=on-failure
Slice=session.slice
```

Reply:
Very odd indeed. I was overriding the nixpkgs input, let me change that and see if we get the same eval:

```
/nix/store/8gww5vdnbxgvd8ian4m7z8rhixb0vsy9-hyprland-0.44.0+date=2024-11-03_d78bc75.drv

/nix/store/p1fx35kj1nw5w6swlz9xwhkbmjbknn55-hyprland-0.44.0+date=2024-11-03_d78bc75
```

It's different to your store path.

Reply:
Oh, I got eventual consistency'd; what idea are you scrapping?

Reply:
We're removing the systemd service and the targets, and will rely on [uwsm](https://github.com/Vladimir-csp/uwsm) to handle systemd integration instead.

The comments in this PR from the past day or so can give more insight into the issue. There's also https://github.com/hyprwm/hyprland-wiki/pull/850.

Reply:
Hm, that's sad because I think a wayland-compositor should ship these rather than some 3rd party session manager but I also haven't grappled with uwsm yet. I'll try it out.

Reply:
I've been hit by many roadblocks while trying out uwsm, so feel free to check [how I did it](https://github.com/fufexan/dotfiles/commit/a6084b9eec934ab9cbb452eb9accd2c20bb38444).

Reply:
I managed to get the basics working quite easily but my setup is quite minimal. Looking at your diff, those changes look sensible in any case.

What I don't like is that there's no longer a target that I can attach hyprland-specific services to like this PR would provide. I wouldn't want my hypridle to run in e.g. GNOME or KDE. Are you supposed to use the `wayland-wm@Hyprland.service` for that?

Reply:
Yes. Though I think the services defined by HM should have an option to change the `After` key, or instruct users to use `systemd.user.services.<service.Unit.After`.

Reply:
> Hm, that's sad because I think a wayland-compositor should ship these rather than some 3rd party session manager but I also haven't grappled with uwsm yet. I'll try it out.

IMHO, there's no point in reinventing the wheel, if a ready-to-use universal solution already exists. Plus, systemd stuff is a rather tricky (as we found out) code that needs to be maintained, fixed etc. Given, how big Hyprland already is and that there are other hypr* ecosystem projects being in development, offloading this part to an already implemented and working solution looks like the best decision.

Reply:
I appreciate wanting to use ready-made solutions but they need to actually solve the problem properly in order to be viable.

Unless I'm missing something, it is not possible in uwsm to create a service that is part of the greater hyprland session but starts after the compositor is up. You cannot make it wantedBy graphical-session because it'd then be wantedBy *any* graphical session, regardless of the compositor. If you made it `After` the compositor service other services `After` the compositor service or the graphical-session wouldn't wait on your service, so that wouldn't work either.  
I'd have expected uwsm to create a target for us that'd allow for this use-case.

Reply:
Nevermind, it does create all sorts of targets. I didn't immediately find them because it appears to generate them dynamically rather than putting them into `/etc/systemd/user/`?!

Reply:
@Vladimir-csp is it feasible to also create a `{name}-session.target` that would be bound to `wayland-wm@{name}.target`?

Reply:
No, as I said, it already does this, I just didn't see it because the units files weren't where I expected them to be.

Reply:
Using `WantedBy=wayland-session@hyprland.desktop.target` while target not yet existing is valid and works, but will result in a (harmless) warning from systemd.

Alternatively there is `systemd-xdg-autostart-condition`, see [examples](https://github.com/Vladimir-csp/uwsm/blob/master/example-units)